commands/project/dev/index.ts

```diff
@@ -21,36 +19,53 @@ import {
   logProfileHeader,
   exitIfUsingProfiles,
 } from '../../../lib/projectProfiles';
+import { commands } from '../../../lang/en';
+import { getAccountId } from '@hubspot/local-dev-lib/config';
 
 const command = 'dev';
-const describe = uiBetaTag(
-  i18n(`commands.project.subcommands.dev.describe`),
-  false
-);
+const describe = uiBetaTag(commands.project.dev.describe, false);
 
 async function handler(
   args: ArgumentsCamelCase<ProjectDevArgs>
 ): Promise<void> {
-  const { derivedAccountId, providedAccountId } = args;
+  const {
+    derivedAccountId,
+    providedAccountId,
+    targetTestingAccount,
+    targetProjectAccount,
+  } = args;
 
   const { projectConfig, projectDir } = await getProjectConfig();
   validateProjectConfig(projectConfig, projectDir);
 
+  const useV3 = useV3Api(projectConfig.platformVersion);
+
   if (!projectDir) {
-    logger.error(
-      i18n(`commands.project.subcommands.dev.errors.noProjectConfig`, {
-        accountId: derivedAccountId,
-        authCommand: uiCommandReference('hs auth'),
-      })
-    );
+    uiLogger.error(commands.project.dev.errors.noProjectConfig);
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  // If either targetTestingAccount or targetProjectAccount is provided, the other must be provided
+  if (
+    (targetTestingAccount && !targetProjectAccount) ||
+    (targetProjectAccount && !targetTestingAccount)
+  ) {
+    uiLogger.error('TODO');
     process.exit(EXIT_CODES.ERROR);
   }
 
-  let targetAccountId = providedAccountId;
+  // Legacy projects do not support targetTestingAccount and targetProjectAccount
+  if (targetTestingAccount && targetProjectAccount && !useV3) {
+    uiLogger.error('TODO');
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  let targetProjectAccountId =
+    providedAccountId || getAccountId(targetProjectAccount);
 
   let profile: HsProfileFile | undefined;
 
-  if (!targetAccountId && useV3Api(projectConfig.platformVersion)) {
+  if (!targetProjectAccountId && useV3) {
```

Comment:
This if condition can never be undefined because `getAccountId(targetProjectAccount)` will fall back to using the default account. I think we want to specifically check if `providedAccountId  || args.targetProjectAccountId` is defined

Reply:
Ohhh good catch. So many of these config related functions are so confusing...can't wait to get rid of them